fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/keyspace/ResolverValidator.java

```diff
@@ -0,0 +1,301 @@
+     */
+    public enum ValidationResult {
+        /**
+         * Entry is valid.
+         */
+        OK,
+
+        /**
+         * Indicates that an entry in the mapping has no corresponding reverse entry.
+         */
+        REVERSE_ENTRY_MISSING,
+
+        /**
+         * Indicates that an entry in the mapping has a corresponding reverse entry, but that reverse entry
+         * belong to a different key than the one that referenced it.
+         */
+        REVERSE_ENTRY_MISMATCH
+    }
+
+    /**
+     * Represents a bad/incorrect entry in the resolvers mapping.
+     */
+    public static class ValidatedEntry {
+        @Nonnull
+        private final ValidationResult result;
+        @Nonnull
+        private final ResolverKeyValue keyValue;
+        @Nonnull
+        private final String reverseValue;
+
+        public ValidatedEntry(@Nonnull final ValidationResult result,
+                              @Nonnull final ResolverKeyValue keyValue) {
+            this(result, keyValue, keyValue.getKey());
+        }
+
+        public ValidatedEntry(@Nonnull final ValidationResult result,
+                              @Nonnull final ResolverKeyValue keyValue,
+                              @Nonnull final String reverseValue) {
+            this.result = result;
+            this.keyValue = keyValue;
+            this.reverseValue = reverseValue;
+        }
+
+        @Nonnull
+        public ValidationResult getValidationResult() {
+            return result;
+        }
+
+        /**
+         * Return the key in the resolver mapping.
+         * @return the key in the resolver mapping
+         */
+        @Nonnull
+        public String getKey() {
+            return keyValue.getKey();
+        }
+
+        /**
+         * The value for the key in the mapping.
+         * @return the value for the key in the mapping
+         */
+        @Nonnull
+        public ResolverResult getValue() {
+            return keyValue.getValue();
+        }
+
+        /**
+         * The key found in the reverse directory associated with the value found in the forward mapping.
+         * For {@link ValidationResult#REVERSE_ENTRY_MISMATCH} this will be a different value than returned
+         * by {@link #getKey()}.
+         *
+         * @return the key found in the reverse directory associated with the value found in the forward mapping
+         */
+        @Nonnull
+        public String getReverseValue() {
+            return reverseValue;
+        }
+
+        @Override
+        public boolean equals(final Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            final ValidatedEntry validatedEntry = (ValidatedEntry)o;
+            return result == validatedEntry.result && keyValue.equals(validatedEntry.keyValue) && Objects.equals(reverseValue, validatedEntry.reverseValue);
```

Comment:
Was `reverseValue` initially `@Nullable`? It seems like this should either use `Objects::equals` on both the `keyValue` and `reverseValue` fields or `Object::equals` for both